src/codeu/chat/server/Model.java

```diff
@@ -67,6 +69,27 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final Store<User, HashSet<Uuid>> userInterestByUser = new Store<>(null);
+  private final Store<User, HashSet<Uuid>> convoInterestByUser = new Store<>(null);
```

Comment:
It would be easier to have the key be `Uuid` instead of the whole `User` object. Then the key would `User.id`.